knative-operator/pkg/common/monitoring.go

```diff
@@ -36,6 +42,36 @@ func SetupMonitoringRequirements(api client.Client, instance mf.Owner) error {
 	return nil
 }
 
+func RemoveOldServiceMonitorResources(namespace string, api client.Client) error {
+	currentSMName := os.Getenv(operatorServiceMonitorNameEnvKey)
+	if currentSMName == "" {
+		currentSMName = operatorServiceMonitorDefaultName
+	}
+	smList := monitoringv1.ServiceMonitorList{}
+	if err := api.List(context.TODO(), &smList, client.InNamespace(namespace)); err != nil {
```

Comment:
I sort of wondered if it'd be simpler to enumerate the items we want to delete explicitly, but not a strong opinion, this approach looks viable as well.

Reply:
I kept this a bit generic not having to define them per release.